app/models/transformation_mapping_item.rb

```diff
@@ -24,4 +30,34 @@ def destination_cluster
       errors.add(:destination, "EMS type of destination cluster or cloud tenant must be in: #{destination_types}")
     end
   end
+
+  def source_datastore
```

Comment:
I think these method names should probably be prefixed with `validate_` to avoid confusion.  I would expect a method named `source_datastore` to return some type of Datastore object.

Reply:
I will rename the validation methods.   I myself don't like `unless` construct.  Let me see what I can do.

Reply:
Can circle back to it during lan-validations